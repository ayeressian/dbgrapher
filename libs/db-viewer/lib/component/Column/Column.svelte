<script context="module" lang="ts">
  export enum Highlight {
    None = 1,
    From,
    To,
  }
</script>

<script lang="ts">
  import type { ColumnFkSchema, ColumnSchema } from "../../schema";
  import { isColumnFk } from "../../schema";
  import pkUrl from "./pk.svg";
  import fkUrl from "./fk.svg";

  export let column: ColumnSchema;
  export let highlight: Highlight;

  export let columnEnter: (column: ColumnFkSchema) => void;
  export let columnLeave: (column: ColumnFkSchema) => void;

  let highlightCssClass = "";
  $: {
    switch (highlight) {
      case Highlight.From:
        highlightCssClass = "highlight-from";
        break;
      case Highlight.To:
        highlightCssClass = "highlight-to";
        break;
      default:
        highlightCssClass = "test";
        break;
    }
  }

  const mouseEnter = () => {
    if (isColumnFk(column)) {
      columnEnter(column);
    }
  };
  const mouseLeave = () => {
    if (isColumnFk(column)) {
      columnLeave(column);
    }
  };
</script>

<style>
  tr {
    border-bottom: 1px solid #bbb;
  }

  td {
    padding: 5px;
    padding-left: 10px;
    padding-right: 10px;
  }

  td.status {
    height: 20px;
    padding: 0;
    flex-direction: row;
  }

  td.status img {
    width: 14px;
    height: 14px;
    margin-left: 7px;
  }

  .highlight-from {
    background-color: lightgreen;
  }

  .highlight-to {
    background-color: lightcoral;
  }
</style>

<tr
  class={highlightCssClass}
  on:mouseenter={mouseEnter}
  on:mouseleave={mouseLeave}
>
  <td class="status">
    <img
      style="display: {column.pk ? 'block' : 'none'}"
      alt="primary key"
      src={pkUrl}
    />
  </td>
  <td class="status">
    <img
      style="display: {isColumnFk(column) ? 'block' : 'none'}"
      alt="foreign key"
      src={fkUrl}
    />
  </td>
  <td>{column.name}</td>
  <td>
    {#if !isColumnFk(column)}
      {column.type}
    {/if}
  </td>
</tr>
