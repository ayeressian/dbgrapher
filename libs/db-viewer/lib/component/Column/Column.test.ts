import type { ColumnSchema } from '../../schema';
import { render, RenderResult, cleanup } from '@testing-library/svelte';
import Column from './Column.svelte';
import { beforeEach, describe, expect, it } from 'vitest';

describe(Column.name, () => {
	let component: RenderResult;

	beforeEach(async () => {
		const columnSchema: ColumnSchema = {
			name: 'test',
			type: 'int'
		};
		cleanup();
		component = render(Column, {
			column: columnSchema
		});
	});
	it('should render properly', () => {
		expect(component).toMatchSnapshot();
	});
});
