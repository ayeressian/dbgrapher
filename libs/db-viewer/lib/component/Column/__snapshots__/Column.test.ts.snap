// Vitest Snapshot v1

exports[`Column$ > should render properly 1`] = `
{
  "component": Column$ {
    "$$": {
      "after_update": [],
      "before_update": [],
      "bound": {},
      "callbacks": {},
      "context": Map {},
      "ctx": [
        {
          "name": "test",
          "type": "int",
        },
        "test",
        [Function],
        [Function],
        1,
        [MockFunction spy],
        [MockFunction spy],
      ],
      "dirty": [
        -1,
      ],
      "fragment": {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": [
        [Function],
      ],
      "on_disconnect": [],
      "on_mount": [],
      "props": {
        "column": 0,
        "columnEnter": 5,
        "columnLeave": 6,
        "highlight": 4,
      },
      "root": <div>
        <tr
          class="test"
        >
          <td
            class="status"
          >
            <img
              alt="primary key"
              src="/lib/component/Column/pk.svg"
              style="display: none;"
            />
          </td>
           
          <td
            class="status"
          >
            <img
              alt="foreign key"
              src="/lib/component/Column/fk.svg"
              style="display: none;"
            />
          </td>
           
          <td>
            test
          </td>
           
          <td>
            int
          </td>
        </tr>
      </div>,
      "skip_bound": false,
      "update": [Function],
    },
    "$$set": [Function],
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div>
      <tr
        class="test"
      >
        <td
          class="status"
        >
          <img
            alt="primary key"
            src="/lib/component/Column/pk.svg"
            style="display: none;"
          />
        </td>
         
        <td
          class="status"
        >
          <img
            alt="foreign key"
            src="/lib/component/Column/fk.svg"
            style="display: none;"
          />
        </td>
         
        <td>
          test
        </td>
         
        <td>
          int
        </td>
      </tr>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
